server/src/main/java/org/elasticsearch/action/get/TransportGetAction.java

```diff
@@ -91,6 +110,16 @@ protected void resolveRequest(ClusterState state, InternalRequest request) {
     protected void asyncShardOperation(GetRequest request, ShardId shardId, ActionListener<GetResponse> listener) throws IOException {
         IndexService indexService = indicesService.indexServiceSafe(shardId.getIndex());
         IndexShard indexShard = indexService.getShard(shardId.id());
+        if (indexShard.routingEntry() == null) {
```

Comment:
Can this happen? Looks like we init it in `IndexShard` constructor and then only update it to non-null values?

Reply:
Yeah, but that looks like both comes from `TransportIndicesStatsAction`, which was done in 2014 and thus on a very different code base. I think we should assume this does not happen, since we assume so in other actions, for instance `TransportGetFromTranslogAction`, `PostWriteRefresh` and many others. Unless you know where it can happen I suggest to remove the new null check, we do not want null-checks all over for things that cannot be null.

Reply:
I agree!